[unicorn/no-array-for-each] rule suggested for pIteration.forEach not array forEach

<!-- ✨ Thanks for reporting a bug! ➡️ Please don't ignore this template -->

<!-- 1️⃣ Explain here what's wrong -->

<!-- 2️⃣ Specify which rule is buggy here and in the title -->

<!-- 3️⃣ Add some examples where the issue appears -->

I've added this plugin and it's suggesting I replace pIteration and not just the built in array's forEach, is this the correct behavior?

```
   95:20  error  Use `for…of` instead of `Array#forEach(…)`           unicorn/no-array-for-each
```

```ts
await pIteration.forEach(plugins, async pluginName => {
    // My other code...
});
```

--- a/rules/no-array-for-each.js
+++ b/rules/no-array-for-each.js
@@ -362,6 +362,8 @@ const ignoredObjects = [
 	'React.Children',
 	'Children',
 	'R',
+	// https://www.npmjs.com/package/p-iteration
+	'pIteration',
 ];
 
 /** @param {import('eslint').Rule.RuleContext} context */

--- a/test/no-array-for-each.mjs
+++ b/test/no-array-for-each.mjs
@@ -11,6 +11,12 @@ test.snapshot({
 		// #1087
 		'React.Children.forEach(children, (child) => {});',
 		'Children.forEach(children, (child) => {});',
+		// #1508
+		outdent`
+			await pIteration.forEach(plugins, async pluginName => {
+				// My other code...
+			});
+		`,
 	],
 	invalid: [
 		// Not fixable

